tests/unit/pypyr/utils/expressions_test.py

```diff
@@ -69,25 +69,3 @@ def test_expr_var_doesnt_exist():
     """Expression raises when variable not found in namespace."""
     with pytest.raises(NameError):
         expressions.eval_string('a', {'b': True})
-
-
-def test_expr_derived_dicts_fail():
-    """Derived dicts fail with eval.
-
-    For reasons best known to eval(), it EAFPs the locals() look-up 1st and if
-    it raises a KeyNotFound error, moves on to globals.
-
-    This means if you pass in something like the pypyr context, the custom
-    KeyNotInContextError the pypyr context raises isn't caught, and thus eval
-    doesn't work.
-
-    Therefore, before pypyr context needs to initialize a new dict(context) in
-    order to be used as the locals arg.
-
-    If this unit test fails, it means the functionality has changed and you can
-    get rid of the redundant dict creation when context get_eval_string invokes
-    expressions eval_string.
```

Comment:
per comment, this means that the dict creation in ./pypyr/context.py ln 247 can be removed.

from `return expressions.eval_string(input_string, dict(self))`
to `return expressions.eval_string(input_string, self)`

Reply:
you are right, I will update pr.